lib/collection/src/shards/local_shard/mod.rs

```diff
@@ -976,30 +976,34 @@ impl LocalShard {
 
         {
             let segments = self.segments().read();
-            for (_, segment) in segments.iter() {
-                let segment_info = segment.get().read().info();
-                if segment_info.segment_type == SegmentType::Special {
-                    status = ShardStatus::Yellow;
-                }
 
-                if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
-                    status = ShardStatus::Red;
-                }
+            if !segments.failed_operation.is_empty() || segments.optimizer_errors.is_some() {
+                status = ShardStatus::Red;
 
                 if let Some(error) = &segments.optimizer_errors {
                     optimizer_status = OptimizersStatus::Error(error.to_string());
                 }
+            } else {
+                let has_special_segments = segments
+                    .iter()
+                    .map(|(_, segment)| segment.get().read().info().segment_type)
+                    .any(segment_type == SegmentType::Special);
+
+                // Special segment means it's a proxy segment and is being optimized, mark as yellow
+                if has_special_segments {
+                    status = ShardStatus::Yellow;
+                }
```

Comment:
Now that I think of it, this would also trigger if we'd just create a snapshot. Snapshotting creates temporary proxy segments.

Let's keep it like this for now though.

Reply:
added a comment